joint_state_controller/test/test_joint_state_controller.cpp

```diff
@@ -0,0 +1,289 @@
+// Copyright 2020 PAL Robotics SL.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+
```

Comment:
I know I'm now *that* guy commenting on your whitespaces but I really don't think it's necessary to bloat the include section with this many newlines. I don't think it's evil to bundle ROS2 includes into one group

Reply:
I don't care either way. But having whitespaces between different package includes is how it's done pretty much everywhere. A few examples:

[joint_state_controller.hpp](https://github.com/ros-controls/ros2_controllers/blob/master/joint_state_controller/include/joint_state_controller/joint_state_controller.hpp)
[joint_trajectory_controller.hpp](https://github.com/ros-controls/ros2_controllers/blob/master/joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp)
[controller_interface.hpp](https://github.com/ros-controls/ros2_control/blob/master/controller_interface/include/controller_interface/controller_interface.hpp)
[controller_manager.hpp](https://github.com/ros-controls/ros2_control/blob/master/controller_manager/include/controller_manager/controller_manager.hpp)

Reply:
Yes I know and it also looks ugly and impractical. The fact that we have bad style in some files shouldn't force us to introduce new code following it. The next time we get to those we can fix this.

Reply:
Understood. Will fix.

Reply:
Done.